spyder/plugins/pythonpath/utils.py

```diff
@@ -18,12 +18,16 @@
 def check_path(path):
     """Check that `path` is not a [site|dist]-packages folder."""
     if os.name == 'nt':
-        pat = re.compile(r'.*lib/(?:site|dist)-packages.*')
+        pattern = re.compile(r'.*(?:l|L)ib/(?:site|dist)-packages.*')
     else:
-        pat = re.compile(r'.*lib/python.../(?:site|dist)-packages.*')
+        pattern = re.compile(
+            r'.*(:?lib|lib64)/'
+            '(?:python|python(\d+)|python(\d+)\.(\d+)|python(\d+)\.(\d+)(\d+))'
```

Comment:
```suggestion
            '(?:python|python\d+|python\d+\.\d+)'
```
I think the `()` are superfluous around `/d+`, and will just create unecessary groups. I think the pattern `python\d+\.\d+\d+` is redundant to `python\d+\.\d+`.

As with my previous comment, I don't think `?:` is necessary, but doesn't hurt either.